python/test/unit/language/test_subprocess.py

```diff
@@ -105,13 +106,15 @@ def test_print(func_type: str, data_type: str, device: str):
 
 
 @pytest.mark.parametrize("func_type", assert_types)
-def test_assert(func_type: str):
+def test_assert(func_type: str, device: str):
     # The total number of elements in the 1-D tensor to assert on.
     N = 128
 
-    os.environ["TRITON_DEBUG"] = "1"
-    proc = subprocess.Popen([sys.executable, assert_path, func_type], stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                            shell=False)
+    proc = subprocess.run(
+        [sys.executable, assert_path, "test_assert", func_type, device],
+        capture_output=True,
+        env={**os.environ, "TRITON_DEBUG": "1"},
+    )
```

Comment:
I took the opportunity to switch to `subprocess.run` which is a cleaner interface. I also simplified the environment handling.